dispatch.go

```diff
@@ -125,6 +124,7 @@ func (d *dispatch) handleCall(calls map[int]*call, c *call) {
 	go func() {
 		select {
 		case <-c.ctx.Done():
+			c.ch <- newCanceledError(c.method, seqid)
```

Comment:
Hmm it's good that this change works with the current tests (with your changes), but I was expecting that there'd be new tests to make sure this behaves like we want (as opposed to e.g. the previous version).

For example, I'd expect a test that does:

```
ctx, cancel := ...
// make transporter drop all packets.
go func() {
  ..., err := foo.RPCFunc(ctx, ...)
  // this code should be reached once cancel is called.
  // check err.
}()
cancel()
// assert that the code above is reached.
// assert that the server doesn't see anything.
```

and then something like:

```
ctx, cancel := ...
// make transporter drop all packets after the RPC call, i.e. block the cancel.
go func() {
  ..., err := foo.RPCFunc(ctx, ...)
  ...
}()
cancel()
// assert that the code above is reached.
// assert that the server sees the RPC but not the cancel, and that the client ignores the reply.
```

and then the other possible orderings of the cancel, reply, etc.

We may already have some of these tests? But not all of them, right?